flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -120,6 +127,40 @@ public void closeOpenChannelConnections(ConnectionID connectionId) {
 		}
 	}
 
+	void connectChannelWithRetry(ConnectingChannel connectingChannel, ConnectionID connectionId)
+		throws IOException, InterruptedException {
+		int count = 0;
+		Exception exception = null;
+		do {
+			try {
+				LOG.info("Connecting to {} at {} attempt", connectionId.getAddress(), count);
+				nettyClient.connect(connectionId.getAddress()).addListener(connectingChannel);
+
+				NettyPartitionRequestClient client = connectingChannel.waitForChannel();
+
+				clients.replace(connectionId, connectingChannel, client);
+				exception = null;
```

Comment:
nit: this seems redundant, because the initialization of this variable is also `null`

Reply:
we should exit the loop here to avoid entering the loop again if no exception

Reply:
Yes, it should return directly.